src/main/java/org/mockito/internal/matchers/text/ArrayIterator.java

```diff
@@ -33,6 +34,9 @@ public boolean hasNext() {
     }
 
     public Object next() {
+        if (!hasNext()) {
```

Comment:
More unit tests is a bigger TODO. Let's create a clear issue to request the community to contribute?